pallets/parachain-staking/src/lib.rs

```diff
@@ -716,6 +760,150 @@ pub mod pallet {
+				for mut candidate in pool[..].iter_mut() {
+					// state should always exist but let's be safe
+					if let Some(mut state) = <CollatorState<T>>::get(&candidate.owner) {
+						if state.stake > new {
+							// safe because we ensure stake > new
+							let delta = state.stake - new;
+							state.stake_less(delta);
+							candidate.amount = candidate.amount.saturating_sub(delta);
+
+							// immediately free delta by setting lock from old to new
+							T::Currency::set_lock(STAKING_ID, &candidate.owner, new, WithdrawReasons::all());
+							<CollatorState<T>>::insert(&candidate.owner, state);
+						}
+					}
+				}
+			});
+
+			// update candidates for next round
+			let (num_collators, num_delegators, _, _) = Self::select_top_candidates();
+
+			<MaxCollatorCandidateStake<T>>::put(new);
+
+			Self::deposit_event(Event::MaxCandidateStakeChanged(old, new));
+			Ok(Some(<T as Config>::WeightInfo::decrease_max_candidate_stake_by(
+				num_collators,
+				num_delegators,
+			))
+			.into())
+		}
+
+		/// Forcedly removes a collator candidate from the CandidatePool and
+		/// clears all associated storage for the candidate and their
+		/// delegators.
+		///
+		/// Prepares unstaking of the candidates and their delegators stake
+		/// which can be withdrawn via `withdraw_unstaked` after waiting at
+		/// least `StakeDuration` many blocks.
+		///
+		/// Emits `CandidateRemoved`.
+		///
+		/// # <weight>
+		/// - The transaction's complexity is mainly dependent on updating the
+		///   `SelectedCandidates` storage in `select_top_candidates` which in
+		///   return depends on the number of `MaxSelectedCandidates` (N).
+		/// - For each N, we read `CollatorState` from the storage.
+		/// ---------
+		/// Weight: O(N + D) where N is `MaxSelectedCandidates` bounded by
+		/// `MaxCollatorCandidates` and D is the number of delegators of the
+		/// collator candidate bounded by `MaxDelegatorsPerCollator`
+		/// - Reads: MaxCollatorCandidateStake, 2 * N * CollatorState,
+		///   CandidatePool, BlockNumber, D * DelegatorState, D * Unstaking
+		/// - Writes: MaxCollatorCandidateStake, N * CollatorState,
+		///   SelectedCandidates, D * DelegatorState, (D + 1) * Unstaking
+		/// - Kills: CollatorState, DelegatorState for all delegators which only
+		///   delegated to the candidate
+		/// # </weight>
+		#[pallet::weight(<T as Config>::WeightInfo::force_remove_candidate(T::MaxCollatorCandidates::get() * T::MaxDelegatorsPerCollator::get()))]
+		pub fn force_remove_candidate(
+			origin: OriginFor<T>,
+			collator: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+			let collator = T::Lookup::lookup(collator)?;
+			let state = <CollatorState<T>>::get(&collator).ok_or(Error::<T>::CandidateNotFound)?;
+			let total_amount = state.total;
+
+			let mut candidates = <CandidatePool<T>>::get();
+			ensure!(
+				candidates.len().saturating_sub(1) as u32 >= T::MinSelectedCandidates::get(),
```

Comment:
Maybe we can use `try_into()` to catch cases in which `usize` is `u64`?

Reply:
fixed in b03f3c6